libraries/helpers.rb

```diff
@@ -0,0 +1,19 @@
+module VarnishCookbook
+  # Helper methods to be used in multiple Varnish cookbook libraries.
+  module Helpers
+    def varnish_version
+      cmd = 'varnishd -V 2>&1'
+      cmd = Mixlib::ShellOut.new(cmd)
+      cmd.environment['HOME'] = ENV.fetch('HOME', '/root')
+
+      begin
+        cmd.run_command
+        Chef::Log.info "varnish version detected: #{cmd}"
+        Chef::Log.info "varnish version detected: #{cmd.stdout}"
+        return cmd.stdout.match(/varnish-([0-9])\./).captures[0]
+      rescue => ex
+        Chef::Log.warn "Unable to run varnishd to get version.\nMessage: #{ex.message}"
```

Comment:
In most cases it won't be critical if the version is not detected. Only if Varnish 4.0 is being installed from varnish-cache. However, I did not generate an exception here because I wanted it to build in why_run mode. If you have it generate an exception, the library won't converge in why_run mode since it will try to detect the varnishd version on your test server.